### PR TITLE
Use k8s 1.17.0 provider

### PR DESCRIPTION
**Pull request by qinqon**
Signed-off-by: Quique Llorente <ellorent@redhat.com>

**What this PR does / why we need it**:

**Special notes for your reviewer**:

**Release note**:

```release-note
NONE
```

### CODE REVIEW

**Comment by qinqon** [comment]
Issue is:
```
Failed to pull image "registry:5000/nmstate/kubernetes-nmstate-handler": rpc error: code = Unknown desc = Error while pulling image: Get http://registry:5000/v1/repositories/nmstate/kubernetes-nmstate-handler/images: dial tcp: lookup registry on [::1]:53: read udp [::1]:42909->[::1]:53: read: connection refused
```
something is not good at the worker.

**Comment by qinqon** [comment]
So, looks like it's master the one that is not able to pull
```
[ellorent@localhost kubernetes-nmstate]$ ./kubevirtci/cluster-up/ssh.sh node02 -- sudo docker pull registry:5000/nmstate/kubernetes-nmstate-handler
2020/01/22 08:31:45 Waiting for host: 192.168.66.102:22
2020/01/22 08:31:45 Connected to tcp://192.168.66.102:22
Using default tag: latest
Trying to pull repository registry:5000/nmstate/kubernetes-nmstate-handler ... 
latest: Pulling from registry:5000/nmstate/kubernetes-nmstate-handler
Digest: sha256:64db4b22e83df0806ba3fb7eaf851425915789659f06c25f8b5561d1a23385c9
Status: Image is up to date for registry:5000/nmstate/kubernetes-nmstate-handler:latest
[ellorent@localhost kubernetes-nmstate]$ ./kubevirtci/cluster-up/ssh.sh node01 -- sudo docker pull registry:5000/nmstate/kubernetes-nmstate-handler
2020/01/22 08:31:53 Waiting for host: 192.168.66.101:22
2020/01/22 08:31:53 Connected to tcp://192.168.66.101:22
Using default tag: latest
Trying to pull repository registry:5000/nmstate/kubernetes-nmstate-handler ... 
Pulling repository registry:5000/nmstate/kubernetes-nmstate-handler
Error while pulling image: Get http://registry:5000/v1/repositories/nmstate/kubernetes-nmstate-handler/images: dial tcp: lookup registry on [::1]:53: read udp [::1]:32776->[::1]:53: read: connection refused
[ellorent@localhost kubernetes-nmstate]$ ./kubevirtci/cluster-up/kubectl.sh get node
NAME     STATUS   ROLES    AGE     VERSION
node01   Ready    master   3m13s   v1.17.0
node02   Ready    <none>   2m46s   v1.17.0
[ellorent@localhost kubernetes-nmstate]$ 
```

**Comment by qinqon** [comment]
There is an issue at kubevirtci 1.17 https://github.com/kubevirt/kubevirtci/issues/238

File: cmd/manager/main.go [hunk 1]
```diff
@@ -172,11 +172,9 @@ func serveCRMetrics(cfg *rest.Config) error {
 	if err != nil {
 		return err
 	}
-	// Get the namespace the operator is currently deployed in.
-	operatorNs, err := k8sutil.GetOperatorNamespace()
-	if err != nil {
-		return err
-	}
+	// This is a cluster scope operator so there is no namespace
```
Comment (phoracek) [inline cmd/manager/main.go]: The operator has a namespace, its CRs do not. This is used to collect metrics of the operator pod (namespaced under nmstate)

File: hack/flush-secondary-nics.sh [hunk 2]
```diff
@@ -4,10 +4,15 @@
 script_dir=$(dirname "$(readlink -f "$0")")
 ssh=$script_dir/../kubevirtci/cluster-up/ssh.sh
 
-# TODO okd4.1
+# TODO okd/ocp
```
Comment (phoracek) [inline hack/flush-secondary-nics.sh]: TODO what?

File: hack/flush-secondary-nics.sh [hunk 3]
```diff
@@ -4,10 +4,15 @@
...
 if [[ "$KUBEVIRT_PROVIDER" =~ k8s ]]; then
-    for n in $(seq 1 $KUBEVIRT_NUM_SECONDARY_NICS); do
-        echo 'Flushing nic $n'
-        $ssh node01 -- sudo nmcli con del "\"Wired connection $n\""
+    for node in $($KUBECTL get nodes --no-headers | awk '{print $1}'); do
```
Comment (phoracek) [inline hack/flush-secondary-nics.sh]: Please explain in a comment what are you doing here and *why*

File: cmd/manager/main.go (hunk 1 above)
Comment (qinqon) [reply cmd/manager/main.go]: Then why this fixed the issue ?

Comment (phoracek) [reply cmd/manager/main.go]: Ah, then I misunderstood what is the purpose

**Pull request merged**